test/e2e/setup.go

```diff
@@ -0,0 +1,367 @@
+	}
+	return true, nil
+}
+
+func randomAddressPrefix() (string, error) {
+	//	10.$((RANDOM & 127)).$((RANDOM & 255)).0/24
+	a, err := rand.Int(rand.Reader, big.NewInt(int64(127)))
+	if err != nil {
+		return "", err
+	}
+	b, err := rand.Int(rand.Reader, big.NewInt(int64(255)))
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("10.%s.%s.0/24", a.String(), b.String()), nil
+}
+
+func (c *clientSet) createVnet(ctx context.Context, clusterName, vnetName string) error {
+	masterPrefix, err := randomAddressPrefix()
+	if err != nil {
+		return err
+	}
+	workerPrefix, err := randomAddressPrefix()
+	if err != nil {
+		return err
+	}
+	t := &arm.Template{
+		Schema:         "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+		ContentVersion: "1.0.0.0",
+		Resources: []*arm.Resource{
+			{
+				Resource: &aznetwork.VirtualNetwork{
+					VirtualNetworkPropertiesFormat: &aznetwork.VirtualNetworkPropertiesFormat{
+						AddressSpace: &aznetwork.AddressSpace{
+							AddressPrefixes: &[]string{
+								"10.0.0.0/9",
+							},
+						},
+						Subnets: &[]aznetwork.Subnet{
+							{
+								SubnetPropertiesFormat: &aznetwork.SubnetPropertiesFormat{
+									AddressPrefix: to.StringPtr(masterPrefix),
+								},
+								Name: to.StringPtr(clusterName + "-master"),
+							},
+							{
+								SubnetPropertiesFormat: &aznetwork.SubnetPropertiesFormat{
+									AddressPrefix: to.StringPtr(workerPrefix),
+								},
+								Name: to.StringPtr(clusterName + "-worker"),
+							},
+						},
+					},
+					Name:     to.StringPtr(vnetName),
+					Type:     to.StringPtr("Microsoft.Network/virtualNetworks"),
+					Location: to.StringPtr(c.location),
+				},
+				APIVersion: apiVersions["network"],
+			},
+		},
+	}
+	c.log.Infof("creating vnet")
+	err = c.deployments.CreateOrUpdateAndWait(ctx, c.rg, "azuredeploy", azresources.Deployment{
+		Properties: &azresources.DeploymentProperties{
+			Template: t,
+			Mode:     azresources.Incremental,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	c.log.Debugf("assigning role to AZURE_FP_CLIENT_ID for vnet")
+	err = c.assignVnetRoleTo(ctx, os.Getenv("AZURE_FP_CLIENT_ID"))
+	if err != nil {
+		return err
+	}
+	c.log.Debugf("assigning role to AZURE_CLUSTER_CLIENT_ID for vnet")
+	err = c.assignVnetRoleTo(ctx, os.Getenv("AZURE_CLUSTER_CLIENT_ID"))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *clientSet) getKubeAdminPassword(ctx context.Context, clusterName string) error {
+	c.log.Info("getting kube admin password")
+	cred, err := c.openshiftclusters.GetCredentials(ctx, clusterName, clusterName)
+	if err != nil {
+		return err
+	}
+	if cred.KubeadminPassword != nil {
+		kubeadminPassword = *cred.KubeadminPassword
+	} else {
+		c.log.Warnf("KubeadminPassword is empty")
+	}
+	return nil
+}
+
+func (c *clientSet) createCluster(ctx context.Context, vnetRG, clusterName string) error {
+	b, err := ioutil.ReadFile("cluster.json")
```

Comment:
I got rid of examples/ in a concurrent commit - can we try not using a json file here?